### ISSUE TITLE
Question: locale property may need to be removed

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-components#2321]
### Question

Should we have components with a `locale` property? (date-picker, input-date-picker, input)

### Helpful Details 

We have a few components using a `locale` property for specifying the locale to use for the component.

However, there's already a global attribute to handle this.  See https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/lang#syntax

Currently, the default for the `locale` prop is set to use the document's lang or "en". like so:

```tsx
/** BCP 47 language tag for desired language and country format */
  @Prop() locale?: string = document.documentElement.lang || "en";
```

However, it should probably just use the `lang` on the host and if not specified, it should fallback to use any `element.closest('[lang]').lang` and maybe even fallback to `navigator.language` if not set on the HTML. See https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language

Does this sound correct?

cc @eriklharper

**Comment by eriklharper** [comment]
I agree.  This is quite similar to the page theme.  It is basically a global that all our components should be aware of.

**Comment by jcfranco** [comment]
This makes sense. We can deprecate `locale` props and remove prior to 1.0.0 since it wouldn't affect the behavior.

**Comment by eriklharper** [comment]
Fixed by #3354
